11-Backbonejs/Views-of-Collections.md

```diff
@@ -0,0 +1,208 @@
+
+
+### About `this.$`
+
+In the `render` method we clear the list using `this.$`.  The method `this.$` performs a jQuery selection for HTML inside `el`.  Using `this.$` we can ensure that our view does not interact with any code outside of it.  You should never use jQuery directly within a view.  Instead you should use `this.$` just as you would jQuery to avoid impacting the DOM outside your view.  
+
+
+### Updating `app.js`
+
+We will adjust our `$(document).ready` handler to use the `TaskView`.
+
+
+```JavaScript
+// app.js
+// imports etc...
+
+import TaskListView from './views/task_list_view';
+
+// ...
+
+$(document).ready(function() {
+  var taskData = [{
+    title: "Create a model",
+    completed: true
+  },
+  {
+    title: "Create a collection",
+    completed: false
+  }];
+  taskList = new TaskList(taskData);
+
+  var taskListView = new TaskListView({
+    model: taskList,
+    template: _.template($('#taskItemTemplate').html()),
+    el: 'main'
+  });
+  taskListView.render();
+});
+```
+
+We create the TaskListView and set it's model to be our taskList collection and set the template attribute.  In this case we start off our View attached to an element in the DOM, in this case our `main` tag.
+
+Lastly we call render on the taskListView.
+
+**Wait The Delete Button Just got broken!!**  Don't worry we'll handle that shortly.  
+
+### Handling Creating New Tasks
+
+Just like we added event handlers in the `TaskView` to handle button clicks we can add an event handler to create a new task and a method to read from the new task form, and add them to the collection.
+
+![Creating New Tasks](images/addButton.png)
+
+
+```javascript
+// views/task_list_view.js
+}, // end of render
+
+events: {
+    'click #add-task': "addTask"
+  },
+  readNewTaskForm: function() {
+    // Get the values from the fields
+    var formTitle = this.$('#title').val();
+    this.$('#title').val('');
+    var formDescription = this.$('#description').val();
+    this.$('#description').val('');
+    var formCompleted = this.$('#completed-checkbox').is(":checked");
+    this.$('#completed-checkbox').prop('checked', false);
+
+    return {
+      title: formTitle,
+      description: formDescription,
+      completed: formCompleted
+    };
+  },
+  addTask: function(e) {
+    var taskData = this.readNewTaskForm();
+    var task = new Task(taskData);
+    this.model.add(task);
+  }
+```
+
+Again this looks very much like what we originally wrote in `app.js`  We did change the code to use `this.$` instead of direct jQuery and called `readNewTaskForm()` as an instance method with `this.`.  
+
+**Note you will need to delete the original event handlers in `app.js`.**
+
+**Gah!** The Add button ALSO doesn't work anymore!
+
+Good point, the delete and add buttons are actually working, and tasks are getting added and removed from the collection, but the problem is we're not re-rending the list.
```

Comment:
I must have taken out the question when I'd split the Views lecture.  It actually is a question, to the students because just adding the "update" event listener makes the **delete** button suddenly work!